tests/templates/test_subroutines/test_select_pauli_rot.py

```diff
@@ -0,0 +1,292 @@
+                    torch.tensor(angles),
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="Y",
+                )
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output_torch = dev.execute(tape[0])[0]
+
+        qs = qml.tape.QuantumScript(
+            [
+                qml.SelectPauliRot(
+                    angles,
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="Y",
+                )
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output = dev.execute(tape[0])[0]
+
+        assert qml.math.allclose(output, output_torch)
+
+    @pytest.mark.tf
+    def test_interface_tf(self):
+        """Test that SelectPauliRot works with tensorflow"""
+
+        import tensorflow as tf
+
+        angles = [1, 2, 3, 4]
+
+        wires = qml.registers({"control": 2, "target": 1})
+        dev = qml.device("default.qubit", wires=3)
+
+        qs = qml.tape.QuantumScript(
+            [
+                qml.SelectPauliRot(
+                    tf.Variable(angles),
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="Y",
+                )
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output_tf = dev.execute(tape[0])[0]
+
+        qs = qml.tape.QuantumScript(
+            [
+                qml.SelectPauliRot(
+                    angles,
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="Y",
+                )
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output = dev.execute(tape[0])[0]
+
+        assert qml.math.allclose(output, output_tf)
+
+    @pytest.mark.jax
+    def test_jax_jit(self):
+        """Test that SelectPauliRot works with jax"""
+
+        import jax
+        import jax.numpy as jnp
+
+        angles = jnp.array([1.0, 2.0, 3.0, 4.0])
+
+        wires = qml.registers({"control": 2, "target": 1})
+        dev = qml.device("default.qubit", wires=3)
+
+        qs = qml.tape.QuantumScript(
+            [
+                qml.SelectPauliRot(
+                    angles,
+                    control_wires=wires["control"],
+                    target_wire=wires["target"],
+                    rot_axis="Y",
+                )
+            ],
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        expected_output = dev.execute(tape[0])[0]
+        generated_output = jax.jit(dev.execute)(tape[0])[0]
```

Comment:
How does this ensure that `SelectPauliRot` is jax jittable since we are jitting `dev.execute` here? 🤔

Reply:
I changed it to test via Qnode 👍